packages/secure_storage/amplify_secure_storage/android/src/main/kotlin/com/amazonaws/amplify/amplify_secure_storage/amplify_secure_storage/EncryptedKeyValueRepository.kt

```diff
@@ -4,40 +4,62 @@
 package com.amazonaws.amplify.amplify_secure_storage.amplify_secure_storage
 
 import android.content.Context
+import android.content.Context.MODE_PRIVATE
 import android.content.SharedPreferences
+import androidx.annotation.VisibleForTesting
 import androidx.security.crypto.EncryptedSharedPreferences
 import androidx.security.crypto.EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV
 import androidx.security.crypto.EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
 import androidx.security.crypto.MasterKeys
+import java.io.File
 
 class EncryptedKeyValueRepository(
     private val context: Context,
     private val sharedPreferencesName: String,
 ) : KeyValueRepository {
 
+    private val sharedPreferences: SharedPreferences by lazy {
+        // Attempt to remove previous SharedPreferences file if not initialized.
+        if (!initializationFlagFile.exists()) {
+            context.getSharedPreferences(sharedPreferencesName, MODE_PRIVATE).edit().clear().commit()
+            initializationFlagFile.createNewFile()
+        }
+        EncryptedSharedPreferences.create(
+            sharedPreferencesName,
```

Comment:
I don't have a good idea of how likely a behavior change like that would be. It seems like that would be a breaking change and would be unlikely, but I'm not sure.

That being said, I don't have any issue with updating it to simply delete the file so I made that update. I updated the tests to still call clear + commit, because removing the file doesn't work in a test env.